Makefile

```
@@ -38,4 +40,31 @@ bootstrap:
 		go get -u $$tool; \
 	done
 
+# the following targets are run in CircleCI as part of the build/test jobs
+
+# build the build image
+build-cross-image:
+	docker build -t cross-image:latest .
+
+# use the pre-built image (with dependencies set-up) to build the binary
+build-in-container: build-cross-image
+	docker run --rm \
+	-v /var/run/docker.sock:/var/run/docker.sock \
+	-v $(CURDIR)/pkg:/go/src/github.com/hashicorp/vault-plugin-database-oracle/pkg \
+	cross-image:latest
+
+# run tests in the build container
+test-in-container: build-cross-image
+	docker run --rm \
+	-v /var/run/docker.sock:/var/run/docker.sock \
+	-v $(CURDIR)/test-results/go:/go/src/github.com/hashicorp/vault-plugin-database-oracle/test-results/go \
+	-e RUN_IN_CONTAINER=1 cross-image:latest \
+    make test-ci
+
+# when running in CirleCI - convert test results to junit xml (for storage)
+test-ci: fmtcheck generate
+	go get -x github.com/jstemmer/go-junit-report
+	CGO_ENABLED=1 go test $(TEST) -timeout=20m -parallel=4 \
+	-v | go-junit-report > test-results/go/go-test-report.xml
```

Running this locally, via `test-in-container`, the pipe to `go-junit-report` means that no `go test` output is sent to the screen (the same is happening in CI) I think it would be useful to print the test results to screen as well as to the junit file. Unfortunately `go-junit-report` doesn't seem to have an option to echo the original text, but as a workaround we can tee the original output to a file, and then do the junit processing later, like this:
```suggestion
	-v | tee test-results/go/go-test-report.raw
	go-junit-report < test-results/go/go-test-report.raw > test-results/go/go-test-report.xml
```

---

Thanks, great idea!